Kubernetes edits + open questions (#1298)
cncf/glossary: content/en/kubernetes.md

Kubernetes is extensible via its [API](/application-programming-interface/), allowing experienced Kubernetes practitioners to leverage its automation capabilities according to their needs.

## Problem it addresses

Infrastructure automation and declarative configuration management have been important concepts for a long time, but they have become more pressing as [cloud computing](cloud-computing/) has gained popularity. 
As demand for compute resources increases and organizations need to provide more operational capabilities with fewer engineers, new technologies and working methods are required to meet that demand. 

## How it helps

Similar to traditional [infrastructure as code](/infrastructure-as-code/) tools, Kubernetes helps with automation but has the advantage of working with containers. 
Containers are more resistant to configuration drift than [virtual](/virtual-machine/) or physical machines. 

Additionally, Kubernetes works declaratively, which means that instead of operators instructing the machine how to do something, they describe — usually as manifest files (e.g., YAML) — what the infrastructure should look like. 

> Kubernetes then takes care of the "how." 

Reviewer: ```suggestion
Kubernetes then takes care of the "how". 
```

## What it is

Kubernetes, often abbreviated as K8s, is an open source container orchestrator. 
It automates the lifecycle of containerized applications on modern infrastructures, functioning as a "datacenter operating system" that manages applications across a [distributed system](/distributed-systems/).

Kubernetes schedules [containers](container/) across [nodes](/nodes/) in a [cluster](/cluster/), bundling several infrastructure resources such as load balancers, persistent storages, and etc. to run containerized applications.

Kubernetes enables automation and extensibility, allowing users to deploy applications declaratively (see below) in a reproducible way. 
Other software solutions in the Kubernetes ecosystem take advantage of that automation and extensibility.

> Kubernetes is easily extendable via its [API](/application-programming-interface/), and its automation helps make other tools more accessible to experienced Kubernetes practitioners.

Reviewer: I tried to make the sentence easier.

```suggestion
Kubernetes is extensible via its [API](/application-programming-interface/) which allows experienced Kubernetes practitioners to utilize the automation capability of Kubernetes in a customized way according to their needs.
```

## What it is

Kubernetes, often abbreviated as K8s, is an open source container orchestrator. 
It automates the lifecycle of containerized applications on modern infrastructures, functioning as a "datacenter operating system" that manages applications across a [distributed system](/distributed-systems/).

Kubernetes schedules [containers](container/) across [nodes](/nodes/) in a [cluster](/cluster/), bundling several infrastructure constructs, sometimes referred to as "primitives." 

> These primitives consist of app instances, load balancers, persistent storage, etc., and are composed into applications.

Author: The original text says, "composed into an application." What does that mean? Building an app?

> These primitives consist of app instances, load balancers, persistent storage, etc., and are composed into applications.

Author: The original text says, "composed into an application." What does that mean? Building an app?

Reviewer: @CathPag 
I think it means
The primitives (resources) are composed into an **environment** to run applications.

> Kubernetes schedules [containers](container/) across [nodes](/nodes/) in a [cluster](/cluster/), bundling several infrastructure constructs, sometimes referred to as "primitives." 
> These primitives consist of app instances, load balancers, persistent storage, etc., and are composed into applications.

Reviewer: My suggestion to simplify sentences is as follows,

```suggestion
Kubernetes schedules [containers](container/) across [nodes](/nodes/) in a [cluster](/cluster/), bundling several infrastructure resources such as load balancers, persistent storages, and etc. to run containerized applications.
```

Kubernetes enables automation and extensibility, allowing users to deploy applications declaratively (see below) in a reproducible way. 
Other software solutions in the Kubernetes ecosystem take advantage of that automation and extensibility.
Kubernetes is easily extendable via its [API](/application-programming-interface/), and its automation helps make other tools more accessible to experienced Kubernetes practitioners.

## Problem it addresses

Infrastructure automation and declarative configuration management have been important concepts for a long time, but they have become more pressing as [cloud computing](cloud-computing/) has gained popularity. 
As demand for compute resources increases and organizations need to provide more operational capabilities with fewer engineers, new technologies and working methods are required to meet that demand. 

> Additionally, the rise of cloud computing along with containerization and organizations busy automating more traditional infrastructure need a mechanism to automate the configuration and deployment of their containers.

Author: what does "organizations busy automating more traditional infrastructure" mean here?

Reviewer: @CathPag 
It is ambiguous to me too. Maybe it is redundant as well considering 

> organizations need to provide more operational capabilities with fewer engineers, new technologies and working methods are required to meet that demand

mentioning a similar issue.

Author: Ok, so we just drop the sentence entirely, right?

Author: I think that's what you meant, so I deleted it. Let me know if that works.
